### PR TITLE
Add a note about gitconfig username and email automatic update

### LINKED ISSUE #22828: Gitconfig is not automatically created
### Describe the bug

Gitconfig is not automatically created on Developer Sandbox / m3 cluster

Screenshot 2024-02-19 at 14 26 19

### Che version

7.81@latest

### Steps to reproduce

1. go to workspaces.openshift.com
2. create a workspace based on github repo e.g. https://github.com/che-incubator/quarkus-api-example
3. SUCCESS - workspace has been started
4. ERROR - there is no Gitconfig created

Screenshot 2024-02-19 at 14 27 40

### Expected behavior

Gitconfig with username / email is created

### Runtime

OpenShift

### Screenshots

_No response_

### Installation method

OperatorHub

### Environment

Amazon

### Eclipse Che Logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by vinokurig**
## What does this pull request change?
Add a note about gitconfig username and email automatic update:
screenshot-0 0 0 0_4000-2024 02 22-14_42_58

## What issues does this pull request fix or reference?
fixes https://github.com/eclipse/che/issues/22828
## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/end-user-guide/pages/mounting-git-configuration.adoc [hunk 1]
```diff
@@ -46,6 +46,11 @@ __<ConfigMap_prepared_in_step_1>__
 EOF
 ----
 
+[NOTE]
+====
+The `user.name` and `user.email` fields can be set automatically from a git provider, connected to {prod-short} by a xref:using-a-git-provider-access-token.adoc[Git-provider access token]. The related data must be set in the provider's user profile page.
```
Comment (ibuziuk) [inline modules/end-user-guide/pages/mounting-git-configuration.adoc]: is it supported for all providers?

File: modules/end-user-guide/pages/mounting-git-configuration.adoc [hunk 2]
```diff
@@ -46,6 +46,11 @@ __<ConfigMap_prepared_in_step_1>__
 EOF
 ----
 
+[NOTE]
```
Comment (ibuziuk) [inline modules/end-user-guide/pages/mounting-git-configuration.adoc]: Let's put the note at the top of the procedure, before pre-requirements

File: modules/end-user-guide/pages/mounting-git-configuration.adoc (hunk 1 above)
Comment (vinokurig) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: yes

Comment (ibuziuk) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: it is not only about access token, but also Git OAuth

Comment (vinokurig) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: It is not related directly to Git OAuth, the idea is to have a token secret either generated by Git OAuth, or manually created PAT secret. So the token documentation reference covers both cases.

**Review by deerskindoll — APPROVED** [review]
lgtm

File: modules/end-user-guide/pages/mounting-git-configuration.adoc (hunk 1 above)
Comment (ibuziuk) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: well, but it is linked with https://eclipse.dev/che/docs/stable/end-user-guide/using-a-git-provider-access-token/ that talks about PAT which is misleading

File: modules/end-user-guide/pages/mounting-git-configuration.adoc (hunk 2 above)
Comment (vinokurig) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: I've changed the configmap block to not to specify just username and email but generalise the gitconfig content:

screenshot-0 0 0 0_4000-2024 02 22-16_31_27

 With this change I think the note suits here more as it specifies that the general gitconfig content can be updated automatically.

File: modules/end-user-guide/pages/mounting-git-configuration.adoc (hunk 1 above)
Comment (vinokurig) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: Mentioned the OAuth source as well.

File: modules/end-user-guide/pages/mounting-git-configuration.adoc [hunk 3]
```diff
@@ -46,6 +46,11 @@ __<ConfigMap_prepared_in_step_1>__
...
+====
+The `user.name` and `user.email` fields can be set automatically to the `gitconfig` content from a git provider, connected to {prod-short} by a xref:using-a-git-provider-access-token.adoc[Git-provider access token] or a token generated via OAuth. The related data must be set in the provider's user profile page.
```
Comment (ibuziuk) [inline modules/end-user-guide/pages/mounting-git-configuration.adoc]:
```suggestion
The `user.name` and `user.email` fields will be set automatically to the `gitconfig` content from a git provider, connected to {prod-short} by a xref:using-a-git-provider-access-token.adoc[Git-provider access token] or a token generated via OAuth, if username and email are set on the provider's user profile page.
```
Please, move this note above before .Prerequisites

File: modules/end-user-guide/pages/mounting-git-configuration.adoc (hunk 2 above)
Comment (ibuziuk) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: @vinokurig this NOTE should be placed at the very beginning, just after the title `Mounting Git configuration`

Comment (vinokurig) [reply modules/end-user-guide/pages/mounting-git-configuration.adoc]: @ibuziuk done

**Review by ibuziuk — APPROVED** [review]

**Comment by ibuziuk** [comment]
@deerskindoll we are good to merge

**Pull request merged**